frontend/src/libs/components/input-file/input-file.tsx

```diff
@@ -0,0 +1,90 @@
+import { FIRST_ARRAY_INDEX } from '#libs/constants/constants.js';
+import { IconColor } from '#libs/enums/enums.js';
+import {
+  getDeepErrorMessage,
+  getValidClassNames,
+} from '#libs/helpers/helpers.js';
+import { useCallback, useFormController } from '#libs/hooks/hooks.js';
+import {
+  type FormControl,
+  type FormFieldErrors,
+  type FormFieldPath,
+  type FormFieldValues,
+} from '#libs/types/types.js';
+
+import { Icon } from '../components.js';
+import styles from './styles.module.scss';
+
+type Properties<T extends FormFieldValues> = {
+  control: FormControl<T, null>;
+  errors: FormFieldErrors<T>;
+  name: FormFieldPath<T>;
+  fileName: string | null;
+  label: string;
+  description: string;
+  onChange?: (file: File) => void;
+};
+
+const InputFile = <T extends FormFieldValues>({
+  control,
+  errors,
+  name,
+  fileName,
+  label,
+  description,
+  onChange,
+}: Properties<T>): JSX.Element => {
+  const { field } = useFormController({ name, control });
+
+  const errorMessage = getDeepErrorMessage(errors[name]);
+  const hasError = Boolean(errorMessage);
+  const displayFile = Boolean(fileName) && !hasError;
+
+  const handleFileChange = useCallback(
+    (event_: React.ChangeEvent<HTMLInputElement>) => {
+      if (event_.target.files?.length) {
+        const currentFile = event_.target.files[FIRST_ARRAY_INDEX] as File;
+
+        if (onChange) {
+          onChange(currentFile);
+        }
+
+        const { name, type, size } = currentFile;
+        field.onChange({ name, type, size });
```

Comment:
You should call it with file
```suggestion
        field.onChange(file);
```

Reply:
Yeah, it's the consequences of same [this](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/pull/213#discussion_r1321975243) problem, I'll try to figure out the solution